cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/DefaultPreviewRunnerModule.java

```diff
@@ -188,6 +194,9 @@ protected void configure() {
 
     bind(PreferencesFetcher.class).toProvider(preferencesFetcherProvider);
     expose(PreferencesFetcher.class);
+
+    bind(PreviewRequestFetcher.class).to(DirectPreviewRequestFetcher.class);
+    expose(PreviewRequestFetcher.class);
```

Comment:
Why need to expose?

Reply:
PreviewManager for standalone creates instances of PreviewRunnerService using injector. PreviewRunnerService requires PreviewRequestFetcher which is why its exposed here.

Reply:
It shouldn't be the PreviewManager to create the runner service in Standalone. Conceptually, the manager and the runner service are two separated services. In distributed, they started in different pod (hence different processes), while in standalone, they are started from Standalone Main independently. We might need a separate service class for starting N number of PreviewRunnerServices, in which that new service is started in StandaloneMain independently of the PreviewManager.

Reply:
I refactored code to separate the PreviewManager and PreviewRunner interfaces. Basically preview manager had method to get the preview runner. And then all HttpHandler requests were served by PreviewRunner - for example getData(), getStatus() etc. I removed those methods from PreviewRunner and moved to PreviewManager in one commit
and added separate guice bindings for them in separate commit.